syntax = "proto3";

service RViz {
  rpc run_code (CodeRequest) returns (Status) {}
  rpc create_instance (CreateInstanceRequest) returns (Status) {}
}

message Empty {
}

message CodeRequest {
  bytes code = 1;
}

message CreateInstanceRequest {
  string name = 1;
  bytes data = 2;
}

message Status {
  enum StatusValue {
    ERROR = 0;
    DONE = 1;
  }
  StatusValue value = 1;
}
